core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -1040,14 +1077,80 @@ protected void discardNode(Entity entity) {
 
     protected void stopAndRemoveNode(Entity member) {
         removeMember(member);
-
         try {
             if (member instanceof Startable) {
-                Task<?> task = member.invoke(Startable.STOP, Collections.<String,Object>emptyMap());
+                Task<?> task = newThrottledEffectorTask(member, Startable.STOP, Collections.<String, Object>emptyMap());
                 task.getUnchecked();
             }
         } finally {
             Entities.unmanage(member);
         }
     }
+
+    protected <T> Task<?> newThrottledEffectorTask(Entity target, Effector<T> effector, Map<?, ?> arguments) {
+        return newThrottledEffectorTask(target, effector, arguments, false);
+    }
+
+    protected <T> Task<?> newThrottledEffectorTask(Entity target, Effector<T> effector, Map<?, ?> arguments, boolean isPrivileged) {
+        final Semaphore permit = childTaskSemaphore;
+        final Task<?> toSubmit;
+        final Task<T> effectorTask = Effectors.invocation(target, effector, arguments).asTask();
+        if (permit != null) {
+            // Acquire the permit now for the privileged task and just queue the effector invocation.
+            // If it's unprivileged then queue a task to obtain a permit first.
+            final String description = "Waiting for permit to run " + effector.getName() + " on " + target;
+            final Runnable obtain = new ObtainPermit(permit, description);
+            if (isPrivileged) {
+                obtain.run();
+                toSubmit = effectorTask;
+            } else {
+                // acquire semaphore here if privileged?
+                Task<?> obtainMutex = Tasks.builder()
```

Comment:
Is it worth moving this and `ObtainPermit` into `Tasks.java`, so that this call would become something like 
```
                Task<?> obtainMutex = Tasks.builder()
                        .description(description)
                        .throttleOn(permit)
                        .build();
```